x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/ConnectorIndexService.java

```diff
@@ -69,14 +71,23 @@ public ConnectorIndexService(Client client) {
     /**
      * Creates or updates the {@link Connector} in the underlying index.
      *
-     * @param docId     The ID of the connector.
-     * @param connector The connector object.
+     * @param request   Request for creating the connector.
      * @param listener  The action listener to invoke on response/failure.
      */
-    public void putConnector(String docId, Connector connector, ActionListener<DocWriteResponse> listener) {
+    public void putConnector(PutConnectorAction.Request request, ActionListener<DocWriteResponse> listener) {
```

Comment:
Just a naming thing:

This can be addressed in a different PR, if you decide to address it 🙏  To me it's not obvious from a caller perspective, what the differences are between `putConnector` and `postConnector`. AFAIU both can be used to create or update a connector, but the sole difference lies in whether an id is auto-generated or not. I think the methods can be renamed to reflect that to the caller explicitly (you can also introduce a flag, but it's probably cleaner to have separate methods with clear names).